### PR TITLE
Add SqsHealthIndicator

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
Add health indicator for SQS listeners that checks if:

- `SimpleMessageListenerContainer` is in running state for all queues mapped with `SqsListener` - even those who do not exist at startup
- all configured queues exist and are reachable.

Fixes: #317

### CODE REVIEW

File: spring-cloud-aws-messaging/pom.xml [hunk 1]
```diff
@@ -45,6 +45,11 @@
 			<groupId>org.springframework</groupId>
 			<artifactId>spring-messaging</artifactId>
 		</dependency>
+		<dependency>
```
Comment (aemruli) [inline spring-cloud-aws-messaging/pom.xml]: @maciejwalkowiak why do we need this dependency here ? I try to keep the boot depdencies out from the modules (only for release 2.0, later I want to add them).

Comment (maciejwalkowiak) [reply spring-cloud-aws-messaging/pom.xml]: Good point. Leftover from previous imolementation. I'll remove it soon.

**Comment by maciejwalkowiak** [comment]
I am not convinced about keeping all health indicators in separate module. Would it be better to keep them close to classes they use to check health? In that case we could limit a bit the public API exposed by these classes - in this case perhaps `getMappedQueueNames` would not need to be public.

**Comment by gmariotti** [comment]
Hi, is there any update on this PR?

**Comment by maciejwalkowiak** [comment]
Spring Cloud GCP keeps all health indicators in `autoconfigure` module. I am thinking if in case of Spring Cloud AWS wouldn't it make more sense to also keep health indicators there and turn current module into `spring-cloud-starter-aws-cloudwatch`. What do you think @eddumelendez?

**Comment by eddumelendez** [comment]
Yes, agree. IMHO, health indicators should be at the `autoconfigure` module since `spring-boot-starter-actuator` is the one that provides the initial infrastructure. So, features can be enabled just adding starters for a specific service.

Regarding to the new module `spring-cloud-starter-aws-cloudwatch`, I think make sense but something that was thinking in `spring-cloud-gcp` is that if the implementation relies on `micrometer` may  using the name`spring-cloud-starter-aws-metrics` make more sense since the `cloudwatch` configuration will be at the `autoconfigure` using mostly micrometer. It's not a big deal the final name though. WDYT?

btw, I just noticed there is no implementation in `spring-boot` but found [this comment](https://github.com/spring-projects/spring-boot/issues/15644#issuecomment-452157980) pointing to this repo.

**Comment by maciejwalkowiak** [comment]
The only reason I think naming it after `cloudwatch` can be safer than `metrics` is that AWS may come up with another metrics related service and it wont be clear what we are referencing to anymore. Similar to what happened with SQS/SNS and our current `messaging` module.

Since this will be a breaking change we would either do it in 3.0 or in 2.3 (considering there will be 2.3).

**Comment by maciejwalkowiak** [comment]
Closing for now, we will revisit this topic for 3.x

**Pull request closed without merging**